sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -43,6 +40,11 @@ private[sql] class SessionState(sparkSession: SparkSession) {
   // Note: These are all lazy vals because they depend on each other (e.g. conf) and we
   // want subclasses to override some of the fields. Otherwise, we would get a lot of NPEs.
 
+  // Since both sharedState and sessionState are lazily initialized in SparkSession,
+  // we need to ensure sharedState are initialized before the sessionState; otherwise,
+  // hive-site.xml might not be loaded when we using it.
```

Comment:
Before this PR, we did it in an implicit way. : )
 https://github.com/apache/spark/blob/2b10ebe6ac1cdc2c723cb47e4b88cfbf39e0de08/sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala#L155